protocol/messenger_communities.go

```diff
@@ -2713,7 +2713,12 @@ func (m *Messenger) UpdateCommunityFilters(community *communities.Community) err
 	publicFiltersToInit := make([]transport.FiltersToInitialize, 0, len(defaultFilters)+len(community.Chats()))
 
 	publicFiltersToInit = append(publicFiltersToInit, defaultFilters...)
-
+	for _, filter := range publicFiltersToInit {
+		_, err := m.transport.RemoveFilterByChatID(filter.ChatID)
```

Comment:
Do we need to cleanup the filters before initializing them? 🤔

Reply:
Also, seems that `filter` might be empty here. (Not `nil` though, so not a panic)

`publicFiltersToInit` is of  `len(defaultFilters) + len(community.Chats())`
But then we only append `defaultFilters` to it, and never append the `community.Chats()`.

should it be just a loop over `defaultFilters`?

Reply:
> Do we need to cleanup the filters before initializing them? 🤔

Yes, because the filter update logic is not very clean and i did not want to hack it there to handle new scenario.  Instead deleting them and re-subscribing seems more cleaner. 

iirc this flow is executed when pubsubTopic is changed for a community which is anyways a breaking config change and should be ok.

Reply:
yep, looks like this is not required for ph-1 changes.